src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolPayment.java

```diff
@@ -39,24 +40,31 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(
         CaseDetails<CaseData, State> details,
         CaseDetails<CaseData, State> detailsBefore
     ) {
-        log.info("Mid-event callback triggered for SolPayment page");
+
+        final Long caseId = details.getId();
+        log.info("Mid-event callback triggered for SolPayment page Case Id: {}", caseId);
 
         CaseData caseData = details.getData();
 
         if (!caseData.getApplication().isSolicitorPaymentMethodPba()) {
-            log.info("Payment method is not PBA for case id {}  :", details.getId());
+            log.info("Payment method is not PBA for case id {}  :", caseId);
             return AboutToStartOrSubmitResponse.<CaseData, State>builder()
                 .data(caseData)
                 .build();
         }
 
-        DynamicList pbaNumbersDynamicList = pbaService.populatePbaDynamicList();
+        try {
+            DynamicList pbaNumbersDynamicList = pbaService.populatePbaDynamicList();
 
-        log.info("DynamicList {}", pbaNumbersDynamicList);
-        caseData.getApplication().setPbaNumbers(pbaNumbersDynamicList);
+            log.info("DynamicList {}, Case Id: {}", pbaNumbersDynamicList, caseId);
+            caseData.getApplication().setPbaNumbers(pbaNumbersDynamicList);
 
-        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .data(caseData)
-            .build();
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .build();
+        } catch (final FeignException e) {
+            log.error("Failed to retrieve PBA numbers for Case Id: {}", caseId);
```

Comment:
worth adding a unit test to cover this scenario ?

Reply:
Added test in commit https://github.com/hmcts/nfdiv-case-api/pull/2240/commits/3277544e04c11d6513c3eaed2b93cc4ac346d47a